.github/actions/build-artifact/action.yaml

```diff
@@ -38,10 +38,12 @@ runs:
         FR_ARTIFACTORY_USER: ${{ inputs.FR_ARTIFACTORY_USER }}
         FR_ARTIFACTORY_USER_ENCRYPTED_PASSWORD: ${{ inputs.FR_ARTIFACTORY_USER_ENCRYPTED_PASSWORD }}
     - name: Push to community snapshot if version contains snapshot
-      if: contains( env.VERSION, 'SNAPSHOT') && inputs.updateMavenServer == true
```

Comment:
I think this would work if you declared updateMavenServer input as a boolean, otherwise you are comparing "true" with true.

https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/